core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/preferences/TitleBlockDialog.java

```diff
@@ -112,22 +118,39 @@ private void createContent(Composite container) {
           IInterpreter vpInterpreter = CompoundInterpreter.INSTANCE.getInterpreterForExpression(contents);
           DSemanticDiagramSpec target = new DSemanticDiagramSpec();
 
-          ContentInstanceContext contentContext = new ContentInstanceContext(target, contents, position);
-          List<ContentProposal> computedProposals;
-          if (vpInterpreter instanceof IProposalProvider) {
-            computedProposals = ((IProposalProvider) vpInterpreter).getProposals(vpInterpreter, contentContext);
+          List<IContentProposal> proposalsList = new ArrayList<IContentProposal>();
+          if (vpInterpreter instanceof CapellaInterpreter) {
+            ContentInstanceContext contentContext = new ContentInstanceContext(target, contents, position);
+            // get all the categories for target and match the command name from category with the command in TitleBlock
+            Set<ICategory> categories = CategoryRegistry.getInstance().gatherCategories(target);
+
+            for (ICategory category : categories) {
+              proposalsList.add(new org.eclipse.jface.fieldassist.ContentProposal(
+                  CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL,
+                      category.getName().trim().replaceAll(" ", "_")),
+                  contentContext.getTextSoFar(), 
+                  null, 
+                  contentContext.getCursorPosition()));
+            }
           } else {
-            computedProposals = new ArrayList<>();
-            final List<IProposalProvider> proposalProviders = ProposalProviderRegistry.getProvidersFor(vpInterpreter);
-            for (IProposalProvider provider : proposalProviders) {
-              computedProposals.addAll(provider.getProposals(vpInterpreter, contentContext));
+            ContentInstanceContext contentContext = new ContentInstanceContext(target, contents, position);

```

Comment:
This can be written only once, outside if (is the same sentence as in the first if).